src/main/java/com/northcoders/makemydayapi/service/RestaurantService.java

```diff
@@ -0,0 +1,130 @@
+package com.northcoders.makemydayapi.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.northcoders.makemydayapi.model.GeoapifyRestaurant;
+import com.northcoders.makemydayapi.model.RestaurantType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@Service
+public class RestaurantService {
+
+    @Value("${api.key}")
+    private String API_KEY;
+
+    // St Paul's Cathedral
+    private final String LATITUDE = "51.5138438";
+    private final String LONGITUDE = "-0.13955";
+
+    // 10 kilometre radius
+    private String radius = "10000";
+    private String resultsLimit = "300";
+
+    @Autowired
+    RestaurantsFilter restaurantsFilter;
+
+    public List<GeoapifyRestaurant> getRestaurants(String type){
+
+        List<GeoapifyRestaurant> restaurants = new ArrayList<>();
+
+        OkHttpClient client = new OkHttpClient.Builder()
+                .readTimeout(0, TimeUnit.SECONDS)
+                .build();
+
+        String endpoint = "https://api.geoapify.com/v2/places?categories=catering&filter=circle:"
+                +LONGITUDE+","+LATITUDE+","+radius+"&limit="+resultsLimit+"&apiKey="+API_KEY;
+
+        Request request = new Request.Builder()
+                .url(endpoint)
+                .header("User-Agent", "OkHttp Headers.java")
+                .addHeader("Content-Type", "application/json")
+                .build();
+
+        String jsonBodyString = null;
+
+        try (Response response = client.newCall(request).execute()) {
+            System.out.println("Service 1) Http request executed");
+            jsonBodyString = response.body().string();
+        } catch (IOException e) {
+            System.err.println("Error making http request " + e.getMessage());
+            return restaurants;
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
Maybe, you can move whole mapping/serializing/converting code to a new different file called as `GeoapifyResponseMapper` under `mapper` folder.

[This](https://github.com/jonquil-NC/make-my-day-api/blob/main/src/main/java/com/northcoders/makemydayapi/mapper/SkiddleResponseMapper.java) is our approach on Skiddle.

Service only calls `toEntity` method of this mapper class and takes an `Activity` back.